src/cli/bin.ts

```diff
@@ -3,6 +3,63 @@ import _capitalize from "lodash/capitalize";
 import fs from "fs-extra";
 import { migrateUIFolder } from "../migrateUIFolder";
 import path from "path";
+import { ContentRecord, DataModel } from "@activeviam/activeui-sdk";
+import { ErrorReport, OutcomeCounters } from "../migration.types";
+import _fromPairs from "lodash/fromPairs";
+
+type MigrationFunction = (
+  legacyUIFolder: ContentRecord,
+  counters: OutcomeCounters,
+  errorReport: ErrorReport,
+  {
+    servers,
+    keysOfWidgetPluginsToRemove,
+    legacyPivotFolder,
+    doesReportIncludeStacks,
+  }: {
+    servers: {
+      [serverKey: string]: {
+        dataModel: DataModel;
+        url: string;
+      };
+    };
+    keysOfWidgetPluginsToRemove?: string[] | undefined;
+    doesReportIncludeStacks: boolean;
+    legacyPivotFolder?: ContentRecord<any> | undefined;
+  },
+) => Promise<[ContentRecord, OutcomeCounters, ErrorReport?]>;
+
+const initialCounters = _fromPairs(
+  ["dashboards", "widgets", "filters", "folders"].map((type) => [
+    type,
+    {
+      success: 0,
+      partial: 0,
+      failed: 0,
+      removed: 0,
+    },
+  ]),
+  // _fromPairs returns a Dictionary.
+  // In this case, the keys used correspond to the attributes of OutcomeCounters.
+) as OutcomeCounters;
+
+const compose =
+  (...migrationFunctions: MigrationFunction[]) =>
+  async (
+    ...contentToMigrate: Parameters<MigrationFunction>
```

Comment:
I find this misleading cause it's not a `ContentRecord`.
It's okay to be abstract and use `...params` here.